filesystem/iso9660.ksy

```diff
@@ -5,206 +5,84 @@ meta:
   xref:
     wikidata: Q815645
   license: CC0-1.0
+  endian: be
+  encoding: ASCII
 doc: |
   ISO9660 is standard filesystem used on read-only optical discs
   (mostly CD-ROM). The standard was based on earlier High Sierra
   Format (HSF), proposed for CD-ROMs in 1985, and, after several
-  revisions, it was accepted as ISO9960:1998.
-
+  revisions, it was accepted as ISO9660:1998.
   The format emphasizes portability (thus having pretty minimal
   features and very conservative file names standards) and sequential
   access (which favors disc devices with relatively slow rotation
   speed).
+doc-ref: |
+  ecma-119 http://www.ecma-international.org/publications/files/ECMA-ST/Ecma-119.pdf
+  susp http://www.ymi.com/ymi/sites/default/files/pdf/Systems%20Use%20P1281.pdf
+  rrip http://www.ymi.com/ymi/sites/default/files/pdf/Rockridge.pdf
+  rras http://www.estamos.de/makecd/Rock_Ridge_Amiga_Specific
+  rrzf https://dev.lovelyhq.com/libburnia/web/wikis/Zisofs
+instances:
+  sector_size:
+    doc-ref: ecma-119 6.1.2
+    value: 0x800
+  volume_descriptor_set:
+    io: _root._io
+    pos: _root.sector_size * 0x10
+    type: volume_descriptor
+    size: sector_size
+    repeat: until
+    repeat-until: _.type == volume_type::volume_descriptor_set_terminator
+enums:
+  volume_type:
+    0x00: boot_record_volume_descriptor
+    0x01: primary_volume_descriptor
+    0x02: supplementary_volume_descriptor
+    0x03: volume_partition_descriptor
+    0xff: volume_descriptor_set_terminator
 types:
-  vol_desc:
-    seq:
-      - id: type
-        type: u1
-      - id: magic
-        contents: "CD001"
-      - id: version
-        type: u1
-      - id: vol_desc_boot_record
-        type: vol_desc_boot_record
-        if: type == 0
-      - id: vol_desc_primary
-        type: vol_desc_primary
-        if: type == 1
-  vol_desc_boot_record:
+  u2bi:
+    doc-ref: ecma-119 7.2.3
     seq:
-      - id: boot_system_id
-        type: str
-        size: 32
-        encoding: UTF-8
-      - id: boot_id
-        type: str
-        size: 32
-        encoding: UTF-8
-  vol_desc_primary:
-    doc-ref: 'http://wiki.osdev.org/ISO_9660#The_Primary_Volume_Descriptor'
+      - id: le
+        type: u2le
+      - id: be
+        type: u2be
+  u4bi:
+    doc-ref: ecma-119 7.3.3
     seq:
-      - id: unused1
-        contents: [0]
-      - id: system_id
-        type: str
-        size: 32
-        encoding: UTF-8
-      - id: volume_id
-        type: str
-        size: 32
-        encoding: UTF-8
-      - id: unused2
-        contents: [0, 0, 0, 0, 0, 0, 0, 0]
-      - id: vol_space_size
-        type: u4bi
-      - id: unused3
-        contents: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
```

Comment:
@GreyCat, it seems we need sequence multiplications and list comprehensions, don't we?